### ISSUE TITLE
Feature Request: Warning for Pattern Matching on Typed Dict

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#9078]
Pyright currently will raise a warning if you try and do an `isinstance` check on a typed dict, however it will not do the same if you try and pattern match on typed dictionaries:

```python
from typing import TypedDict


class User(TypedDict):
    name: str
    age: int


class AdminUser(User):
    is_admin: bool


def process_users(users: list[User | AdminUser]):
    for user in users:
        match user:
            case AdminUser(): # No warning, runtime error
                print("admin user")
            case User():
                print("standard user")



isinstance(User(name="Max", age=25), User) # Raises a warning
```

[Pyright Playground](https://pyright-play.net/?pyrightVersion=1.1.382&pythonVersion=3.14&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgCYAiSAxjAFD1UA2AhgM6tQCqrxIAFEQRlKNAJQAuWlGlQUzCMXFRWMEFJnM0izCjoMW7KAEFSEVN158LICeulJWAfWanUSgEZgwjerVLFgKARwKmJ2RwBXHhBWPijeViVGBxgAbWsoAB9jVxRrAF1bGShgXCh4kB1y6MS7YqgIZhgqAAtq3kl6rqgqNmIcszzovgkoAGIoADkwKAB3ZhAUVDQAGigQCN0sft5wNW6DoJBUGD4AIhVmFFIF0naQM9E6g96eLmGiw67gk-OXQfuj18tAcqEuKFCVmGcgUAF4zgBZZgADzOa00xFhACYAKyiNbWUTjKAAJWYDjCUGYcwWS3QQA)

**Comment by erictraut** [comment]
I think this is a reasonable enhancement request. 

You may have noticed this already, but mypy also doesn't include a check for this condition.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.383.
